crates/runtime/src/accelerated_table/refresh.rs

```diff
@@ -0,0 +1,683 @@
+                })?;
+        let df = ctx
+            .sql(
+                format!(
+                    "SELECT CAST({column} AS timestamp) AS a FROM accelerated_{} ORDER BY a DESC LIMIT 1",
+                    self.dataset_name.clone()
+                )
+                .as_str(),
+            )
+            .await.context(super::FailedToQueryLatestTimestampSnafu)?;
+        let result = &df
+            .collect()
+            .await
+            .context(super::FailedToQueryLatestTimestampSnafu)?;
+
+        let Some(result) = result.first() else {
+            return Ok(None);
+        };
+
+        let array = result.column(0)
+            .as_any()
+            .downcast_ref::<TimestampNanosecondArray>()
+            .context(super::FailedToFindLatestTimestampSnafu {
+                reason: "Failed to get latest timestamp during incremental appending process due to time column is unable to cast to timestamp",
+            })?;
+
+        if array.is_empty() {
+            return Ok(None);
+        }
+
+        let mut value = array.value(0) as u128;
+
+        let schema = &self.accelerator.schema();
+        let Ok(accelerated_field) = schema.field_with_name(&column) else {
+            return Err(super::Error::FailedToFindLatestTimestamp {
+                reason: "Failed to get latest timestamp due to time column not specified"
+                    .to_string(),
+            });
+        };
+
+        if let arrow::datatypes::DataType::Int8
+        | arrow::datatypes::DataType::Int16
+        | arrow::datatypes::DataType::Int32
+        | arrow::datatypes::DataType::Int64
+        | arrow::datatypes::DataType::UInt8
+        | arrow::datatypes::DataType::UInt16
+        | arrow::datatypes::DataType::UInt32
+        | arrow::datatypes::DataType::UInt64 = accelerated_field.data_type()
+        {
+            match refresh.time_format.clone() {
+                Some(TimeFormat::UnixMillis) => {
+                    value *= 1_000_000;
+                }
+                Some(TimeFormat::UnixSeconds) => {
+                    value *= 1_000_000_000;
+                }
+                _ => (),
+            }
+        };
+
+        Ok(Some(value))
+    }
+
+    async fn get_full_or_incremental_append_update(
+        &self,
+        overwrite_timestamp_in_nano: Option<u128>,
+    ) -> super::Result<DataUpdate> {
+        let dataset_name = self.dataset_name.clone();
+        let refresh = self.refresh.read().await;
+        let filter_converter = self.get_filter_converter(&refresh);
+
+        tracing::info!("[refresh] Loading data for dataset {dataset_name}");
+        status::update_dataset(dataset_name.as_str(), status::ComponentStatus::Refreshing);
+        let refresh = refresh.clone();
+        let mut filters = vec![];
+        if let Some(converter) = filter_converter.as_ref() {
+            if let Some(timestamp) = overwrite_timestamp_in_nano {
+                filters.push(converter.convert(timestamp, Operator::GtEq));
```

Comment:
Shouldn't this be Gt, not GtEq?

Reply:
No, as it could have data with same timestamp.

> duplication handling. This is a common problem for key-based incremental append. Some vendor (stitch) uses Greater/Equal operation to fetch new dates which prefer data duplication over incompleteness. And right now the code is doing so. Further enhancement can build more comprehensive record batch comparison to reduce the data duplication.

Listed in the description.

Reply:
Sure but there could also be new data with a lower timestamp

Reply:
yeah, it is tricky one. We can tune it later or even expose the configure for this.

Reply:
Opt for Gt